engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamProcessor.java

```diff
@@ -37,6 +37,31 @@
 import java.util.function.Function;
 import org.slf4j.Logger;
 
+/*
++-------------------+                                  +-----------------+
+|                   |                                  |                 |
+|   ActorStarting   |     ---------------------------> |    Actor close  | <-------------------------------
+|                   |     |          |                 |                 |                                 |
++-------------------+     |          |                 +-----------------+                                 |
+         |                |          |                                                                     |
+         v                |          |                                                                     |
++-------------------+     |    +-----------+        +-------------+        +-----------------+      +------------+
+|                   |------    |           |        |             |        |                 |      |            |
+|  Create Reader    | ------>  |   Replay  |----->  |   Replay    |------> | Create writer   | ---> |   Process  |
+|                   |          |           |        |   Completed |        |                 |      |            |
++-------------------+          +-----------+        +-------------+        +-----------------+      +------------+
+                                      |                                            |                      |
+                                      |                                            |                      |
+                                      |                                            |                      |
+                                      v                                            |                      |
+                                +-------------+                                    |                      |
+                                |   Actor     |                                    |                      |
+                                |   Failed    |  <---------------------------------------------------------
+                                |             |
+                                +-------------+
+
+https://textik.com/#f8692d3c3e76c699
```

Comment:
:+1: Cool :D

Reply:
Maybe we add as well ActorStarted after reader creation, to make it clear it is before started :shrug:

Reply:
:+1: